Source/p_saveg.c

```diff
@@ -2638,6 +2682,8 @@ void P_UnArchiveSpecials (void)
           pusher_t *pusher = Z_Malloc (sizeof(pusher_t), PU_LEVEL, NULL);
           saveg_read_pusher_t(pusher);
           pusher->thinker.function = T_Pusher;
+          // can't convert from index to pointer, old save version
+          if (pusher->source == NULL)
```

Comment:
braces, please

Reply:
and we need a check if source is still NULL afterward and error out if pusher type is either push or pull